tests/e2e_logging.py

```diff
@@ -1467,6 +1467,47 @@ def test_rekey(network, args):
     return network
 
 
+@reqs.description("Test UDP echo endpoint")
+@reqs.at_least_n_nodes(1)
+def test_udp_echo(network, args):
+    primary, _ = network.find_primary()
+    udp_interface = primary.host.rpc_interfaces["secondary_rpc_interface"]
+    host = udp_interface.public_host
+    port = udp_interface.public_port
+    LOG.info(f"Testing UDP echo server at {host}:{port}")
+
+    server_address = (host, port)
+    buffer_size = 1024
+    test_string = b"Some random text"
+
+    with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as s:
+        s.sendto(test_string, server_address)
+        recv = s.recvfrom(buffer_size)
+
+    text = recv[0]
+    LOG.info(f"Testing UDP echo server received {text}")
+    assert text == test_string
+
+
+def run_udp_tests(args):
+    # Register secondary interface as an UDP socket on first node
+    udp_interface = infra.interfaces.make_secondary_interface("udp")
+    args.nodes[0].rpc_interfaces.update(udp_interface)
```

Comment:
We should update this on all nodes, not just node 0 (assumption that `args.nodes[0]` constructs `primary` found later isn't safe).

Reply:
Right, this is left-over from a time when I was creating the nodes by hand. Now I should just use `get_primary` like the other tests (as I did initially).

Reply:
You can't use `get_primary` here - the network doesn't exist yet (just the `args` that specify how it should be created), and the primary may change during execution. Instead we want to make sure every node gets a UDP interface, by updating the spec for every node:

```suggestion
    for node in args.nodes:
        node.rpc_interfaces.update(udp_interface)
```

Reply:
Done